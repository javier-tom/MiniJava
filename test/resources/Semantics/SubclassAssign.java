class Main {
    public static void main(String[] args) {
        System.out.println(new SubclassAssign().run());
    }
}

class SubclassAssign {
    public int run() {
        A a;
        B b;
        B bad;
        int x;
        MethodParamTest t;

        a = new A();
        b = new B();
        t = new MethodParamTest();
        a = b;
        bad = a;

        // Now test it in method parameters
        x = t.testA(a);
        x = t.testB(b);
        x = t.testA(b); // should be valid
        x = t.testB(a); // should error out
        return x;
    }
}

class MethodParamTest {
    public int testA(A a) {
        return 5;
    }

    public int testB(B b) {
        return 5;
    }
}

class A {}

class B extends A {}
