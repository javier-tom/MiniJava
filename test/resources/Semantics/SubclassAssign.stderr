Error on line 19: cannot assign A to a B
Error on line 25: expected type B in parameter 1 of the call to testB. Got type A instead.